app/src/main/java/app/revanced/integrations/patches/ReturnYouTubeDislikePatch.java

```diff
@@ -1,39 +1,63 @@
 package app.revanced.integrations.patches;
 
+import android.text.Spanned;
+
 import java.util.concurrent.atomic.AtomicReference;
 
 import app.revanced.integrations.returnyoutubedislike.ReturnYouTubeDislike;
+import app.revanced.integrations.settings.SettingsEnum;
+import app.revanced.integrations.utils.LogHelper;
 
 /**
  * Used by app.revanced.patches.youtube.layout.returnyoutubedislike.patch.ReturnYouTubeDislikePatch
  */
 public class ReturnYouTubeDislikePatch {
 
     /**
-     * Called when the video id changes
+     * Injection point
      */
     public static void newVideoLoaded(String videoId) {
         ReturnYouTubeDislike.newVideoLoaded(videoId);
     }
 
     /**
+     * Injection point
+     *
      * Called when a litho text component is created
      */
     public static void onComponentCreated(Object conversionContext, AtomicReference<Object> textRef) {
         ReturnYouTubeDislike.onComponentCreated(conversionContext, textRef);
     }
 
     /**
+     * Injection point
+     *
+     * Called when a Shorts dislike Spannable is created
+     */
+    public static Spanned onShortsComponentCreated(Spanned dislike) {
+        return ReturnYouTubeDislike.onShortsComponentCreated(dislike);
+    }
+
+    /**
+     * Injection point
+     *
      * Called when the like/dislike button is clicked
      *
      * @param vote -1 (dislike), 0 (none) or 1 (like)
      */
     public static void sendVote(int vote) {
-        for (ReturnYouTubeDislike.Vote v : ReturnYouTubeDislike.Vote.values()) {
-            if (v.value == vote) {
-                ReturnYouTubeDislike.sendVote(v);
-                return;
+        try {
+            if (!SettingsEnum.RYD_ENABLED.getBoolean()) return;
```

Comment:
This check is already being done here:

https://github.com/revanced/revanced-integrations/blob/076289eb1bf171843eeb45691bf032930dced85e/app/src/main/java/app/revanced/integrations/returnyoutubedislike/ReturnYouTubeDislike.java#L297

It might be desired to remove it from there or here.